golang/go-guestbook/kubernetes-manifests/guestbook-backend.deployment.yaml

```diff
@@ -26,3 +26,7 @@ spec:
           value: "8080"
         - name: GUESTBOOK_DB_ADDR
           value: guestbook-mongodb:27017
+    initContainers:
+        - name: mongo-init
+          image: busybox:1.28
+          command: ['sh', '-c', 'echo $(date) "> checking for envt var for mongo-service"; until nslookup mongo-service; do echo $(date) "> waiting for mongo"; sleep 2; done; echo $(date) "> got envt var for mongo-service"']
```

Comment:
1. `mongo-service` name sounds incorrect.

2. I've never ever seen in my life a Kubernetes deployment that waits on another service like this.
  The expected behavior is to crash or retry inside the app, not outside in the deployment manifest.

Reply:
I agree with @ahmetb. Plus, this won't work. When the script ends, the mongo pod has an IP but no one can connect to it yet. I've pushed a PR to fix that in nodeJs. The idea, as @ahmetb says, is to retry multiple time the connection to the db.

Reply:
@dgageot We added handling for crashing a few times into "Cloud Code: Deploy" action. So I'm not sure if retries are necessary. We can rely on Kubernetes restarts here, and a lot of Kubernetes users do too.

When the mongodb.Ping() method causes backend to crash during the initialization, the "Cloud Code: Deploy" demo still works (which is why I'm not sure what's the need for this fix).

Reply:
What if you deploy this sample not using Cloud Code?
I prefer the code to reflect best practices in addition to the extension.

Reply:
It deploys just fine maybe after restarting 2 times, which is perfectly acceptable in kubernetes. You introduce more failure modes and "look ready" while you're retrying whereas you arent.